### DISCUSSION TITLE
Importing after updating module creates duplicates

### DISCUSSION THREAD

**Discussion by tbergstedt** [fvtt-fria-ligan/forbidden-lands-foundry-vtt#144] [Q&A] [answered]
I have a FoundryVTT installation ongoing since 0.7.9, now "updated" to 0.8.8. What I had to do get 0.8.8 running was to copy the data to a completely new setup, separate from the old one and existing in parallel (the old one of course being shut down at the moment). 

Now when I update the game system and the FBL modules, I get the "Begin importing" dialogs at startup. Even though it says in the settings window that "previously existing entries will be overwritten", what actually happens is that I get duplicates of all items, characters, journal entries, etc. This is of course annoying, and I want to avoid it - but I still want to do the import in order to get any new macros, updates, etc. How can I do that?

**Comment by aMediocreDad** [comment, marked as answer]
Hi, 

I could probably have been clearer here in regards to going from 0.7 to 0.8 and module importing. From 0.8 onwards entities will be overwritten, if you went from 0.7 to 0.8 it would create duplicates like was the behaviour back in 0.7. The reason the Welcome prompt appeared again was because I had to change how I checked whether the module was imported or not. 

On 0.8 you can get the new macros by deleting the old ones manually (unfortunately no other way since I cannot delete them without risking deleting user made macros), and either open the module import dialogue again through settings and selecting only macros, or by going to compendiums finding the macro compendium and clicking import all.

To get other updates there is unfortunately no good solution to it beyond deleting the old stuff (making sure to keep whatever you want to keep like the main map and such), and reimporting. 

Note that future importing WILL overwrite **your** changes to the content (because I have no way of knowing what has been changed by a user).

Unrelated to the question:
I converted this to a discussion since it is not an issue with the system. For future reference there's [a separate repository](https://github.com/aMediocreDad/fbl-content-modules) used for tracking issues with the content module (it is tied to the [Bug Reporter Module](https://foundryvtt.com/packages/bug-reporter).

**Comment by tbergstedt** [comment]
Thanks for the reply. Since I know I have backups, I think I'll remove all previous content and re-import everything. Then I know I'll have the latest versions of everything, and it will make importing from later versions easier (overwriting previous content is a real improvement, if you ask me). 

One more reason why you never should mix own content with that from modules!

And I fully understand why you moved this from Issues, given that explanation.
